specs/kvstore/WriterFencing.fizz

```diff
@@ -0,0 +1,145 @@
+
+    
+    #db.rs\db\open_with_fp_registry() does below
+    atomic func Open():
+        self.state = "initializing"
+        load_manifest_result = self.LoadLatestManifest()
+        if load_manifest_result != SUCCESS:
+            return
+        self.WriteFencingWal()
+
+    #flush.rs\DbInner\flush() is the only method that increments next_wal on write.
+    atomic func WriteWal():
+        store = wal_store
+        write_status = store.ConditionalWrite(self.next_wal, self.manifest.writer_epoch)
+        if write_status == SUCCESS:
+            self.next_wal = self.next_wal + 1
+        else:
+            self.state = "fenced"
+        return write_status
+
+    #db.rs\DbInner\fence_writers()
+    atomic func WriteFencingWal():
+        wal = wal_store.List()
+        if(len(wal) > 0):
+            self.next_wal = wal[len(wal) - 1] + 1
+        else:
+            self.next_wal = 1
+
+        store = manifest_store
+        while self.state == "ManifestUpdated":
+            write_status = self.WriteWal()
+            if write_status == SUCCESS:
+                self.state = "opened"
+                self.was_ever_open = 1
+                break;
+            else:
+                manifests = store.List()
+                recent_manifest = store.Read(manifests[len(manifests) - 1])
+                # if there is a new writer, set current to failed.  Otherwise, try with next wal.
+                if self.manifest.writer_epoch < recent_manifest.writer_epoch:
+                    self.state = "fenced"
+                    break
+                else:
+                    self.next_wal = self.next_wal + 1
+                
+action Init:
+    wal_store = ObjectStore()
+    manifest_store = ObjectStore()
+    writers = []
+    writer_ids = range(0, NUM_WRITERS)
+    for i in range(0,NUM_WRITERS):
+        writers.append(SlateDBWriter())
+
+atomic fair action WriterOpen:
+    writer_id = any writer_ids
+    current_writer = writers[writer_id]
+    if current_writer.state == "waiting":
+        current_writer.Open()
+
+always assertion NotMoreThanOneWriterInWritableState:
+    writable_writer_count = 0
+    wal_list = wal_store.List()
+    for i in range(0, NUM_WRITERS):
+        # writer state must be opened, and next wal slot be available in object store for the write to succeed.
+        if writers[i].state == "opened" and writers[i].next_wal not in wal_list:
+            writable_writer_count +=1
+    return writable_writer_count <=1
+
+# After a new writer writes wal, old writer cannot write its wal. Assert by reading writer epoch of sequential wal.
+always assertion OlderWriterCannotAddWal:
+    wal_list = wal_store.List()
+    previous_wal_writer_epoch = 0
+    for wal in wal_list:
+        wal_writer_epoch = wal_store.Read(wal)
+        if wal_writer_epoch < previous_wal_writer_epoch:
+            return False
+        previous_wal_writer_epoch = wal_writer_epoch
+    return True
+
+exists assertion MultipleWritersOpened:
```

Comment:
@jayaprabhakar  has been helping me with the model offline. He quickly [added](https://github.com/fizzbee-io/fizzbee/commit/2b597391572f4e1c2e83d4e5d822105c3745ab94) this new feature `exists assertion`, it lets us validate that certain paths are possible.